ui-v2/app/mixins/with-event-source.js

```diff
@@ -0,0 +1,16 @@
+import Mixin from '@ember/object/mixin';
+
+export default Mixin.create({
+  reset: function(exiting) {
+    if (exiting) {
+      Object.keys(this).forEach(prop => {
+        if (this[prop] && typeof this[prop].close === 'function') {
+          this[prop].close();
+          // ember doesn't delete on 'resetController' by default
+          delete this[prop];
+        }
+      });
+    }
+    return this._super(...arguments);
+  },
+});
```

Comment:
The above uses the controller-lifecycle PR (https://github.com/hashicorp/consul/pull/5056) to automatically cleanup any 'closeable' things, in this case EventSources